[Merge] HC-129-MyPage To dev

kwondns opened the pull request:
# 📝 개요(이미지 첨부 선택)
*<!-- 이슈에 대해 어떻게, 무엇을, 왜 수정했는지 작성해주세요. -->*

## 4가지 하위 브랜치 작업 후 Merge
- #58 
- #59 
- #61 
- #62 

- MyPage
  > MyPage 관련 레이아웃 및 aside 제작 및 현재 경로에 맞는 active 상태 스타일 적용
  
  <img width="389" alt="image" src="https://github.com/user-attachments/assets/eca24714-b8f9-4b30-9eec-c90ec0912200">

  - MyActivity #59 
  > 내 활동 페이지 (계정 설정으로 이동, 현재 설정된 프로필 및 수정 모달 생성) 내가 쓴 게시글 및 댓글 미구현

  ![image](https://github.com/user-attachments/assets/993e7390-3fdb-472e-a2bc-4667e760e571)

     - MyAccount #61 

     > 내 계정 설정(아바타와 닉네임, 비밀번호 변경) 회원 탈퇴 기능, OAuth 여부에 따른 다른 UI
  
    ![스크린샷 2024-08-02 17 34 52](https://github.com/user-attachments/assets/d9a5769c-b0b4-4565-89a0-2adc8ff53340)

    - MyActivityUpdateProfile #62 
    > 사용자 프로필 변경 모달 생성 및 프로필 수정 기능

    ![image](https://github.com/user-attachments/assets/ce85839f-d962-4c88-b528-f17d68a478cb)

  - MyBookmark #58 
  > 유저의 북마크 현황 및 페이지네이션, 필터링 구현, 하우스 카드 컴포넌트 생성

  ![image](https://private-user-images.githubusercontent.com/155633601/352014013-633d82f3-4f37-4f2b-a5df-9fe090936b43.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1ODkyMDgsIm5iZiI6MTcyMjU4ODkwOCwicGF0aCI6Ii8xNTU2MzM2MDEvMzUyMDE0MDEzLTYzM2Q4MmYzLTRmMzctNGYyYi1hNWRmLTlmZTA5MDkzNmI0My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwMlQwODU1MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYjI2Y2VjZmMyYWQ3YWJhZmM2OWJjMWI2NzM3YTdkZmRiY2VmNTQzMDYwMGFjNTBjMGVlMjczZjM3N2I4ZjczJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qDdiYEynH2LKG6CvUSWYPLvm9DH0kQ7AF1w0JMxgVhc)

# ✅ PR 유형
어떤 변경 사항이 있나요?

- [X] 새로운 기능 추가
- [X] 버그 수정
- [X] CSS 등 사용자 UI 디자인 변경
- [X] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

# 📌 PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다.  
- [X] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).
- [ ] 모든 팀원이 PR에 대한 리뷰를 진행했습니다.
- [X] 이 코드 변경 사항은 기존 기능에 영향을 주지 않나요?
- [X] 팀원이 이해할 수 있도록 충분한 주석 및 코드 일관성을 준수했나요? (변수명, 함수명, 클래스명, 가독성 등)

# 💁‍♂️ PR 추가 정보
*<!-- PR과 관련된 기타 정보(E.G 개선사항, 리뷰어가 봐주었으면 하는 점)가 있다면 여기에 적어주세요. -->*

# 🙏 팀원요청 사항
<!-- PR관련하여 팀원들의 추가 작업이 필요한 경우 작성해주세요 -->

kwondns committed:
🚀 [Feat]: My Page 라우터 생성

- mypage 하위 activity, bookmark, account 경로 추가
- 추후 업데이트 예정인 페이지 경로 생성
- about 임시 페이지에 해당 경로 이동 버튼 추가

kwondns committed:
🚀 [Feat]: MyPage Sidebar 생성

- 추후 업데이트 예정인 페이지를 포함하여 Sidebar 생성
- 각각의 아이템 클릭시 링크로 이동
- 기본적인 레이아웃 작업

kwondns committed:
🚑️ [Fix]: MyPage 중복 라우터 제거

- alarm 관련된 중복 라우터 제거

kwondns committed:
🔨 [Chore]: CherryPick File

- BadgeIcon
- profileDetailInfo

kwondns committed:
🚀 [Feat]: 라우터 설정

- mypage/activity 페이지 지정

kwondns committed:
🚀 [Feat]: MyActivity 관련 fetch 추가

- React-Query의 queryOptions를 사용하여 쿼리 객체 생성
- MyActivity Page에서 해당 쿼리 객체를 사용하여 fetch
- query시 user(session)이 필요하지만 초기화 전에 null을 반환
- 따라서 user가 존재 할 때만 요청을 보내도록 enabled: !!user 추가
- 해당 옵션을 통해 user의 정보가 있을 경우만 fetch

kwondns committed:
🚀 [Feat]: MyActivity UI 적용

- 기본적인 UI 적용

- TODO
- Badge 최신화 후 Hover효과 등 제거 필요
- Icon 추가후 수정 필요

kwondns committed:
🚀 [Feat]: Router 설정

- MyBookmark 관련 라우터 설정
- mypage/bookmark

kwondns committed:
🚀 [Feat]: useMyBookmark 생성

- 로그인한 유저의 bookmark 목록을 가져오는 훅 생성
- react-query의 query-option을 사용
- user_bookmark 테이블 참조하여 데이터 패칭

kwondns committed:
🚀 [Feat]: 하우스 북마크 목록 Fetch

- useMyBookmarkList를 활용하여 하우스 북마크목록
- 임시 BookmarkTemplate 적용

kwondns committed:
🚀 [Feat]: 아이콘 mini-heart 추가

- 북마크 페이지의 mini-heart svg 추가

kwondns committed:
🔨 [Chore]: CherryPick

- BadgeIcon 및 profile 정보 변환 객체

kwondns committed:
🚀 [Feat]: bookmark 관련 아이콘 추가

- page-next, page-prev, search svg 추가

kwondns committed:
🚀 [Feat]: Pagination 컴포넌트 추가

- [currentPage, setCurrentPage]
- totalPage
- 위 2가지를 프로퍼티로 생성
- 현재 위치에 따라 페이지의 목록이 변경
- 1, ..., 4,5,6, ..., 10 와 같은 형태
- 현재 위치가 4이하일 경우 1~5 표시
- 현재 위치가 최대 페이지 - 3 이상일 경우 연결되서 표기
- 전체 페이지가 6 이하일 경우 ... 없이 표기
- 페이지가 없을 경우 제거

kwondns committed:
🔨 [Chore]: Patch  From 375b79a7

- House Type 정의파일 필요하여 패치

kwondns committed:
🚀 [Feat]: House Card 타입 정의

- 하우스 카드에서 사용되는 타입 정의
- 기존 하우스 타입에서 사용되는 타입만 분리
- undefined 제거

kwondns committed:
🔨 [Chore]: Patch House Type

kwondns committed:
🔨 [Chore]: Patch 181f4448

- Badge 스타일 관련 패치

kwondns committed:
🚀 [Feat]: Bookmark Store 생성

- PageAtom -> 목록에서 페이지 번호
- CurrentTabAtom -> 현재 탭 위치 (하우스, 라운지, 게시물)
- HouseFilterAtom -> 하우스 필터링 문자

kwondns committed:
🚀 [Feat]: HouseCard 컴포넌트 생성

- HouseCard 컴포넌트 생성
- House 이미지, 어필, 집 유형, 금액 유형, 지역, 도시, 기간, 보증금, 월세
- 위 데이터를 토대로 하우스 카드 컴포넌트

kwondns committed:
🚀 [Feat]: House Bookmark Fetch

- House Bookmark 리스트 패칭
- House Bookmark 개수 패칭
- 개수를 통해 전체 페이지의 개수를 생성

kwondns committed:
🚀 [Feat]: MyBookmark Page 제작

- useQueries를 사용
- 북마크된 하우스 리스트 9개 단위, 북마크 하우스 총 개수를 패칭
- 패칭 후 MyBookmarkTemplate props로 전달

kwondns committed:
🚀 [Feat]: MyBookmark Template 제작

- Tab을 통해서 하위 템플릿 변경
- 0일 경우 House 리스트 템플릿
- 1일 경우 Lounge 리스트 템플릿
- 2일 경우 Article 리스트 템플릿
- 각각의 템플릿에 house, lounge, article props로 전달
- 각각 전달된 props로 렌더링 예정

kwondns committed:
🚀 [Feat]: House Bookmark template

- 위치 검색(필터링)과 HouseCard 컴포넌트로 구성
- 하우스가 없을 경우(필터링 결과 포함) 없다는 문구 출력
- 위치 검색시 엔터 입력후 필터링 동작

kwondns committed:
🚀 [Feat]: Article, Lounge template

- 게시물 및 라운지 서비스 준비중 문구가 담긴 템플릿 제작

kwondns committed:
🚑️ [Fix]: page-next 아이콘 스타일 수정

- stroke 뿐 아니라 fill 요소로도 구성된 svg
- fill 스타일도 추가하여 전체적인 색상을 유지하도록 함

kwondns committed:
🔨 [Chore]: Icon 관련 패치

- 아이콘 관련 asset, type, const 적용

kwondns committed:
♻️ [Refactor]: Icon 수정

- Patch된 아이콘에 따라 자기소개 영역 아이콘 적용

kwondns committed:
🔨 [Chore]: Patch 181f4448

- Badge 스타일 관련 패치

kwondns committed:
💄 [Design]: Badge patch 후 적용

- MyActivity 페이지의 Badge hover 관련 효과 제거

kwondns committed:
Merge pull request #59 from house-mate-connect/HC-130-MyActivity

[Merge] HC-130-My-Activity to HC-129-MyPage

kwondns committed:
🔨 [Chore]: Patch 755a7e91

- Google, Kakao logo patch

kwondns committed:
🚑️ [Fix]: 사용자 정보가 부족할 경우 에러 발생

- 유저 프로필을 완성하지 않을 경우 map 사용 시 에러 발생
- map 사용시 데이터가 있을 경우에만 하도록 수정

kwondns committed:
🚀 [Feat]: icon 추가

- edit_avatar, kakaotalk-logo-text 추가

kwondns committed:
💄 [Design]: FormItem.Password visible 아이콘 수정

- labelname 존재 여부에 따라 visible 위치가 변경되어 수정

kwondns committed:
🚀 [Feat]: Avatar Supabase Storage 경로

- Avatar에 사용되는 src의 경로가 Supabase Storage일 경우 경로 추가
- 'images/avatar/' 로 시작할 경우 supabase Storage라 가정
- 앞에 환경변수 VITE_SUPABASE_AVATAR_STORAGE_URL 추가

kwondns committed:
🚀 [Feat]: MyAccount Router 적용

- Router.tsx 추가
- MyPageLayout에서 account 경로일 경우 activity 내 활동 활성화

kwondns committed:
🚀 [Feat]: MyAccount 타입 정의

- avatar: 아바타 이미지 파일
- nickname: string 닉네임 변경
- password: string 암호 변경
- passwordConfirm: string 암호 확인

- 전부 optional로 있을 경우에만 적용되도록 함
- passwordValidate 타입을 참조하기 위해 export 로 변경

kwondns committed:
🚀 [Feat]: MyAccount 페이지 생성

- SessionAtom을 통해 Props로 template로 전달

- 변경버튼을 누를시 Typography -> TextField로 변환
- 아바타, 닉네임, 패스워드 중 입력된 데이터만 수정
- 계정삭제 클릭시 ConfirmModal 생성
- ConfirmModal의 메시지에 줄바꿈이 존재하여 style 수정

kwondns committed:
🚑️ [Fix]: 불필요 className 제거

- MyActivity.template 불필요 코드 제

kwondns committed:
🚀 [Feat]: useMyAccount 생성

- MyAccount 페이지 관련 훅 생성
- useMyAccountUpdate
- 계정 설정 변경하는 훅
- avatar, nickname, password 중 입력된 데이터로 계정 설정

- avatar 파일 존재시 Supabase Storage 저장
- images 버킷의 avatar/${id}/avatar라는 경로로 이미지 저장
- upsert를 통해 단일 아바타 파일로 유지

- useDeleteMyAccount 생성
- 직접적으로 auth 스키마를 참조 불가
- user 테이블의 현재 아이디를 가진 데이터 제거
- DB Trigger를 통해 user의 데이터가 제거 될 경우 Auth 테이블도 제거
- 제거 성공시 로그아웃 과정 및 페이지 이동
- User가 삭제된 경우 supabase.auth.signout 불가
- 따라서 수동으로 로그아웃 과정 발생
- SessionAtom, UserAtom null 초기화
- localStorage 관련 데이터 제거
- navigate로 sign/in 페이지 이동

kwondns committed:
🚑️ [Fix]: updateUser 파라미터 수정

- confirmPassword 데이터 제거 후 updateUser 전달

kwondns committed:
🚀 [Feat]: Email 계정만 비밀번호 표시

- Email로 로그인된 계정만 비밀번호 변경이 노출되도록 수정
- Google, Kakao와 같은 OAuth는 패스워드가 없음

kwondns committed:
Merge branch 'refs/heads/dev' into HC-134-MyActivity-Update-Profile

kwondns committed:
🚀 [Feat]: Modal QueryClient 내부로 이동

- GlobalModal의 위치를 QueryClientProvider 내부로 이동
- ProfileModifyModal에서 Query 사용 예정

kwondns committed:
♻️ [Refactor]: Regions 타입 export

- Profile 업데이트 시 사용하기 위하여 타입 내보내기

kwondns committed:
🚀 [Feat]: ProfileModifyModal 기본 설정 추가

- GlobalModal ProfileModify 및 타입 추가
- GlobalModal.store 에 관련 Atom 생성

kwondns committed:
🚀 [Feat]: ProfileModifyModal 생성

- MyActivity Template에서 유저 정보를 전달
- 수정 버튼 클릭 시 ProfileModifyModal 생성
- GlobalModal의 경우 최상위 컴포넌트에서 생성되어 초기 렌더링 시 데이터가 null
- 따라서 useEffect의 dependency에 userInfo를 추가
- 각 데이터를 토대로 form의 기본값으로 설정

kwondns committed:
🚀 [Feat]: useUpdateProfile 작성

- SignUpProfileType과 같은 payload, 수정을 위한 유저의 id 2가지가 필요
- 해당 데이터를 user_lifestyle, user_mate_style, user_looking_house
- 3가지의 테이블에 맞는 데이터로 업데이트 수행
- Promise.All을 통해 동시 수행
- 그 결과 중 하나라도 error가 있을 경우 에러 발생
- mutation 성공 시 closeProfileModifyModal로 모달 종료

kwondns committed:
🚑️ [Fix]: ProfileModifyModal Toast 수정

- ContainerId를 제거하여 정상적으로 에러 토스트 생성
- 미제거시 해당 ToastContainer를 참조하기에 ToastContainer를 추가하거나 제거
- 그중 제거하는 방법을 택하여 다른 토스트와 동일하게 적용되도록 함

kwondns committed:
Merge branch 'HC-129-MyPage' into HC-131-MyBookmark

kwondns committed:
Merge pull request #58 from house-mate-connect/HC-131-MyBookmark

[Merge] HC-131-my-bookmark to HC-129-MyPage

kwondns committed:
Merge branch 'HC-129-MyPage' into HC-132-MyAccount

kwondns committed:
Merge pull request #61 from house-mate-connect/HC-132-MyAccount

[Merge] HC-132-MyAccount to HC-129-MyPage

kwondns committed:
Merge pull request #62 from house-mate-connect/HC-134-MyActivity-Update-Profile

[Merge] HC-134-My-Activity-Update-Profile To HC-129-MyPage

kwondns committed:
🚑️ [Fix]: Conflict Resolve 후 수정

- resolve 후 생긴 오타를 수정

kwondns committed:
🚀 [Feat]: MyAccount 페이지 취소 기능 추가

- 취소시 mypage/activity 페이지로 이동

kwondns committed:
🚑️ [Fix]: MyBookmark 페이지 수정

- house 데이터가 [] 로 넘어올 경우 에러 발생
- house?.length로 조건을 변경하여 해결

rlwjd31 commented on src/components/pages/MyAccount.tsx:
```diff
@@ -0,0 +1,11 @@
+  const session = useRecoilValue(SessionAtom);
+  if (!session) return <Typography.Head2>데이터가 없습니다.</Typography.Head2>;
+  return <MyAccountTemplate session={session} />;
+}
```
이 부분 아래와 같이 protected Router에서 처리했으니 MyAccount와 연관된 컴포넌트가 없으면 
삭제하는 게 좋을 듯

```typescript

function ProtectedRouter({ children }: ProtectedRouterType) {
  ... //

  if (isInitializingSession) return <h1>로그인 정보 확인 중</h1>;

    ... //
  }

  return children;
}
```

rlwjd31 commented on src/components/pages/MyActivity.tsx:
```diff
@@ -0,0 +1,13 @@
+
+export default function MyActivity() {
+  const user = useRecoilValue(UserAtom);
+  const { data } = useQuery(userInfoQuery(user));
```
user fetching하는 query는 chat에서도 있으니 merge후 같이 의논 후 refactoring 진행 필요해 보임
TODO로 merge후 refactoring 작성해주세여

rlwjd31 commented on src/components/templates/MyActivity.template.tsx:
```diff
@@ -0,0 +1,299 @@
+  children,
+  className,
+}: {
+  children: ReactNode | ReactNode[];
```
ReactNode는 렌더링할 수 있는 모든 유형의 자식 요소를 다루는 타입으로 알고 있는데 ReactNode[]와 같이 배열로도 작성한 이유가 어떤 오류가 발생해서임?

rlwjd31 commented on src/components/templates/MyActivity.template.tsx:
```diff
@@ -0,0 +1,299 @@
+  const profileModifyContext: ProfileModifyModalState = {
+    isOpen: true,
+    type: 'ProfileModify',
+    userInfo: {
```
아래와 같이해도 동일한 property 참조는 피할 수 있을 거 같음

```typescript
userInfo: {
      ...user.user_mate_style,
      ...user.user_lifestyle,
      ...user.user_looking_house,
      ...user,
    },
```

rlwjd31 commented on src/components/templates/MyBookmarkArticle.template.tsx:
```diff
@@ -0,0 +1,9 @@
+import Typography from '@/components/atoms/Typography';
```
props로 text를 받아 범용성 있게 만드는게 좋을 듯(다른  준비 중인 서비스가 있어 문구가 바뀔 수 있기 때문)

rlwjd31 commented on src/constants/iconSVG.tsx:
```diff
@@ -272,6 +291,75 @@ const IconSVG: { [key in IconType]: FC<SVGProps<SVGSVGElement>> } = {
       {...others}
     />
   ),
```
mini 버전은 className props로 내려줘서 해결할 수 있는데
새로 추가한 이유가 있는지???

New-Ha replied:
mini버전 추가는 제가 했슴당! 
props 받도록 수정하기 전에 추가했던거 pull해와서 형식만 바꾸고 써서 그래요!
나중에 리팩토링하면서 사이즈만 달라서 필요없는 건 삭제할 예정입니다!

rlwjd31 reviewed (commented):
TODO만 작성해주면 될 거 같아여
고생했슴다👍

kwondns committed:
🔨 [Chore]: Apply Patch

- 37225011
- queryKey Constants 생성

kwondns committed:
♻️ [Refactor]: [House] QueryKeys Factory Pattern 적용

- useMyBookmark.ts

- USER_BOOKMARK (BOOKMARK 목록)
- user, bookmark, list, userId, page, filter

- USER_BOOKMARK_HOUSE_COUNT (HOUSE BOOKMARK 개수)
- user, bookmark, house, count, userId, filter

- useUserInfo.ts

- USER_INFO (USER 정보)
- user, info, userId

kwondns committed:
♻️ [Refactor]: review 수정

kwondns committed:
Merge branch 'dev' into HC-129-MyPage
